
Backport of wpa_supplicant integration from Buildroot mainline.

Not needed for recent BR.

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-05-10 16:43:00.000000000 +0200
+++ buildroot/package/Config.in	2010-05-10 16:43:00.000000000 +0200
@@ -227,6 +227,7 @@
 source "package/wget/Config.in"
 endif
 source "package/wireless-tools/Config.in"
+source "package/wpa_supplicant/Config.in"
 
 endif
 
Index: buildroot/package/wpa_supplicant/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/wpa_supplicant/Config.in	2010-05-10 16:43:00.000000000 +0200
@@ -0,0 +1,32 @@
+config BR2_PACKAGE_WPA_SUPPLICANT
+	bool "wpa_supplicant"
+	help
+	  WPA supplicant for secure wireless networks
+
+	  http://hostap.epitest.fi/wpa_supplicant/
+
+config BR2_PACKAGE_WPA_SUPPLICANT_EAP
+	bool "Enable WPA with EAP"
+	depends on BR2_PACKAGE_WPA_SUPPLICANT
+	help
+	  Enable WPA with EAP (IEEE 802.1X)
+
+config BR2_PACKAGE_WPA_SUPPLICANT_OPENSSL
+	bool "Use openssl TLS (recommended)"
+	depends on BR2_PACKAGE_WPA_SUPPLICANT && BR2_PACKAGE_WPA_SUPPLICANT_EAP
+	select BR2_PACKAGE_OPENSSL
+	help
+	  Use OpenSSL's TLS implementation instead of internal libtommath.
+
+config BR2_PACKAGE_WPA_SUPPLICANT_CLI
+	bool "Install wpa_cli binary"
+	depends on BR2_PACKAGE_WPA_SUPPLICANT
+	help
+	  Install wpa_cli command line utility
+
+config BR2_PACKAGE_WPA_SUPPLICANT_PASSPHRASE
+	bool "Install wpa_passphrase binary"
+	depends on BR2_PACKAGE_WPA_SUPPLICANT
+	help
+	  Install wpa_passphrase command line utility
+
Index: buildroot/package/wpa_supplicant/wpa_supplicant.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/wpa_supplicant/wpa_supplicant.mk	2010-05-11 15:02:38.000000000 +0200
@@ -0,0 +1,84 @@
+#############################################################
+#
+# wpa_supplicant
+#
+#############################################################
+
+WPA_SUPPLICANT_VERSION = 0.6.10
+WPA_SUPPLICANT_SITE = http://hostap.epitest.fi/releases
+WPA_SUPPLICANT_LIBTOOL_PATCH = NO
+WPA_SUPPLICANT_INSTALL_TARGET_OPT = DESTDIR=$(TARGET_DIR) install
+WPA_SUPPLICANT_DEPENDENCIES =
+WPA_SUPPLICANT_CONFIG = $(WPA_SUPPLICANT_DIR)/wpa_supplicant/.config
+WPA_SUPPLICANT_SUBDIR = wpa_supplicant
+WPA_SUPPLICANT_TARGET_BINS = wpa_cli wpa_supplicant wpa_passphrase
+WPA_SUPPLICANT_DBUS_SERVICE = fi.epitest.hostap.WPASupplicant
+
+ifeq ($(BR2_PACKAGE_LIBNL),y)
+	WPA_SUPPLICANT_DEPENDENCIES += libnl
+endif
+
+ifeq ($(BR2_PACKAGE_WPA_SUPPLICANT_OPENSSL),y)
+	WPA_SUPPLICANT_DEPENDENCIES += openssl
+endif
+ifeq ($(BR2_PACKAGE_DBUS),y)
+	WPA_SUPPLICANT_DEPENDENCIES += pkgconfig dbus
+	WPA_SUPPLICANT_MAKE_ENV = \
+		PKG_CONFIG_SYSROOT_DIR="$(STAGING_DIR)"	\
+		PKG_CONFIG_PATH="$(STAGING_DIR)/usr/lib/pkgconfig"
+endif
+
+$(eval $(call AUTOTARGETS,package,wpa_supplicant))
+
+$(WPA_SUPPLICANT_TARGET_CONFIGURE):
+	cp $(WPA_SUPPLICANT_DIR)/wpa_supplicant/defconfig $(WPA_SUPPLICANT_CONFIG)
+	echo "CFLAGS += $(TARGET_CFLAGS)" >>$(WPA_SUPPLICANT_CONFIG)
+ifeq ($(BR2_PACKAGE_DBUS),y)
+	echo "CFLAGS += -I$(STAGING_DIR)/usr/include/dbus-1.0/" >>$(WPA_SUPPLICANT_CONFIG)
+	echo "CFLAGS += -I$(STAGING_DIR)/usr/lib/dbus-1.0/include/" >>$(WPA_SUPPLICANT_CONFIG)
+endif
+	echo "CC = $(TARGET_CC)" >>$(WPA_SUPPLICANT_CONFIG)
+	$(SED) "s/\/local//" $(WPA_SUPPLICANT_DIR)/wpa_supplicant/Makefile
+ifneq ($(BR2_PACKAGE_WPA_SUPPLICANT_EAP),y)
+	$(SED) "s/^CONFIG_EAP_*/#CONFIG_EAP_/g" $(WPA_SUPPLICANT_CONFIG)
+	echo "CONFIG_TLS=none" >>$(WPA_SUPPLICANT_CONFIG)
+else
+ifeq ($(BR2_PACKAGE_WPA_SUPPLICANT_OPENSSL),y)
+	echo "CONFIG_TLS=openssl" >>$(WPA_SUPPLICANT_CONFIG)
+else
+	echo "CONFIG_TLS=internal" >>$(WPA_SUPPLICANT_CONFIG)
+	echo "CONFIG_INTERNAL_LIBTOMMATH=y" >>$(WPA_SUPPLICANT_CONFIG)
+endif
+endif
+ifeq ($(BR2_PACKAGE_DBUS),y)
+	echo "CONFIG_CTRL_IFACE_DBUS=y" >>$(WPA_SUPPLICANT_CONFIG)
+endif
+ifeq ($(BR2_PACKAGE_LIBNL),y)
+	echo "CONFIG_DRIVER_NL80211=y" >>$(WPA_SUPPLICANT_CONFIG)
+endif
+	touch $@
+
+$(WPA_SUPPLICANT_HOOK_POST_INSTALL):
+ifneq ($(BR2_PACKAGE_WPA_SUPPLICANT_CLI),y)
+	rm -f $(TARGET_DIR)/usr/sbin/wpa_cli
+endif
+ifneq ($(BR2_PACKAGE_WPA_SUPPLICANT_PASSPHRASE),y)
+	rm -f $(TARGET_DIR)/usr/sbin/wpa_passphrase
+endif
+ifeq ($(BR2_PACKAGE_DBUS),y)
+	$(INSTALL) -D \
+	  $(WPA_SUPPLICANT_DIR)/wpa_supplicant/dbus-wpa_supplicant.conf \
+	  $(TARGET_DIR)/etc/dbus-1/system.d/wpa_supplicant.conf
+	$(INSTALL) -D \
+	  $(WPA_SUPPLICANT_DIR)/wpa_supplicant/dbus-wpa_supplicant.service \
+	  $(TARGET_DIR)/usr/share/dbus-1/system-services/$(WPA_SUPPLICANT_DBUS_SERVICE).service
+endif
+	touch $@
+
+$(WPA_SUPPLICANT_TARGET_UNINSTALL):
+	$(call MESSAGE,"Uninstalling")
+	rm -f $(addprefix $(TARGET_DIR)/usr/sbin/, $(WPA_SUPPLICANT_TARGET_BINS))
+	rm -f $(TARGET_DIR)/etc/dbus-1/system.d/wpa_supplicant.conf
+	rm -f $(TARGET_DIR)/usr/share/dbus-1/system-services/$(WPA_SUPPLICANT_DBUS_SERVICE).service
+	rm -f $(WPA_SUPPLICANT_TARGET_INSTALL_TARGET) $(WPA_SUPPLICANT_HOOK_POST_INSTALL)
+
